scripts/pre-commit

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+set -e
+
+echo "Running Pre-Commit"
+
+FILES=`git diff --staged --name-only | grep '.*\.go' | cat`
+
+[ -z "$FILES" ] && exit 0
```

Comment:
It might be good to print an exit message here: `&& echo "Pre-commit Vacuously Passed" && exit 0`